javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -191,16 +191,20 @@ public boolean isOrdered() {
         return ofAll(naturalComparator(), elements);
     }
 
+    @SuppressWarnings("unchecked")
     public static <T> PriorityQueue<T> ofAll(Comparator<? super T> comparator, Iterable<? extends T> elements) {
         Objects.requireNonNull(elements, "elements is null");
-
-        int size = 0;
-        Seq<Node<T>> forest = List.empty();
-        for (T value : elements) {
-            forest = insert(comparator, value, forest);
-            size++;
+        if (elements instanceof PriorityQueue && ((PriorityQueue) elements).comparator == comparator) {
```

Comment:
:+1: 
This means the exact same instance of the comparator, not the same logic, right?
Seems ok to me :)